src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -133,51 +169,53 @@ public int GetHashCode(DirectoryInfo value)
             }
         }
 
-        private bool TryAddAssemblyFromManifest(string pathToManifest, string manifestFolder, List<string> assembliesList, ElementLocation location)
+        private bool TryAddAssemblyManifestFromXml(string pathToManifest, string manifestFolder, List<SdkResolverManifest> manifestsList, ElementLocation location)
         {
             if (!string.IsNullOrEmpty(pathToManifest) && !FileUtilities.FileExistsNoThrow(pathToManifest)) return false;
 
-            string path = null;
-
+            SdkResolverManifest manifest = null;
             try
             {
                 // <SdkResolver>
                 //   <Path>...</Path>
+                //   <ResolvableSdkPattern>(Optional field)</ResolvableSdkPattern>
                 // </SdkResolver>
-                var manifest = SdkResolverManifest.Load(pathToManifest);
+                manifest = SdkResolverManifest.Load(pathToManifest);
 
                 if (manifest == null || string.IsNullOrEmpty(manifest.Path))
                 {
                     ProjectFileErrorUtilities.ThrowInvalidProjectFile(new BuildEventFileInfo(location), "SdkResolverDllInManifestMissing", pathToManifest, string.Empty);
                 }
 
-                path = FileUtilities.FixFilePath(manifest.Path);
+                manifest.Path = FileUtilities.FixFilePath(manifest.Path);
```

Comment:
Nit: Do we ever need `Path` without it being ran through `FileUtilities.FixFilePath`? Looks like it could be file-path-fixed during the `Load`.

Reply:
Reading through the code I get the feeling that a lot of the path-fixing logic could be done inside `SdkResolverManifest.Load(pathToManifest);`

Reply:
I did not have a goal to improve this function actually. These changes were only needed so that I divide loading into two steps - getting a valid manifest for all of them and loading them. I agree - it seems that we never need to have a path without it being "fixed". Should I spend any time to move the path fixing logic one level below? I am not sure, here is not a bad place also.